app/Models/Beatmapset.php

```diff
@@ -547,9 +547,13 @@ public function setApproved($state, $user)
 
         $this->save();
 
-        $this
-            ->beatmaps()
-            ->update(['approved' => $this->approved]);
+        $beatmaps = $this->beatmaps();
+
+        if (isset($beatmapIds) && empty(array_diff($beatmapIds, $beatmaps->get()->modelKeys()))) {
```

Comment:
`$beatmaps->whereKey($beatmapIds)->count() === count($beatmapIds)`. And probably should be before saving the beatmapset state and throw error if doesn't match? Seems weird completely ignoring away the parameter if it contains invalid value.

Reply:
Sure, what error should be thrown?

Reply:
`InvariantException` should be fine.